client/src/helpers/getMarkdownLink.js

```diff
@@ -0,0 +1,9 @@
+import { config } from 'config'
+import formatStringToIdName from './formatStringToIdName'
+// Returns a link of the given route and section name
+// a route can be grabbed by visiting the page (e.g., 'terms-of-use', refine.bio/terms-of-use)
+// all the linkable sections are listed in config.markdownLinkable[route]
+export default (route, section) =>
+  config.markdownLinkable[route].includes(section)
+    ? `/${route}#${formatStringToIdName(section)}`
+    : ''
```

Comment:
> Maybe I'm missing something but I don't see an immediate benefit to checking if it is a known section and if not, returning an empty string for the link.

This check was added to ensure that the given `section` (e.g., `Access to and Use of Content`) exists in the given `route` (e.g.,  the `terms-of-use` page) and is included in the config file. So that if it doesn't exist, then the link won't be clickable (empty string) and we can immediately know about it (to prevent linking to the wrong part of the page etc). 

For instance, an invalid link won't be generated if a developer using this helper makes an error and passes `Eligibilty` instead of `Eligibility`. Let me know if that won't be necessary (maybe being too cautious?).

Also I was thinking about adding FE test for this `getMarkDownLink` helper to perform config match.  

<hr/>

In your comment https://github.com/AlexsLemonade/scpca-portal/pull/555#discussion_r1494691128 about the cofig file [config/markdown-linkable.js](https://github.com/AlexsLemonade/scpca-portal/pull/555/files/89cc2a298a80ce97c24e141c2691bf8b37c4cce4#diff-a2bbf6b565d45efbf7a8bb8dbfe70babb2b400331cd3a3c423d4efa54d57f3ac):

> Please look at my notes on `getMarkdownLink` as well for this. We probably want to add the relationship to a markdown file and then the path to the section.
>
> This can be initially bootstrapped automatically. I think that it should be edited manually.
>
> I would expect to see something like this:
> ```js
> export default {
 > accessToAndUseOfContent: {
 >   file: 'terms-of-use.md',
 >   path: '/terms-of-use#access-to-and-use-of-content'
 > }
>}
>```
>
> Again this would allow us to write a test that would pull the configs and then test the individual links without rendering the entire page. Also we wouldn't have to worry about "hardcoded" value that are harder to test.

Initially, we discussed about creating the config file using the text content of each section (e.g., `Eligibility`, `Account Registration`, `Access to and Use of Content`). So currently, we use the most recent markdown file to populate a list of linkable sections at runtime using JS without hand coding those text content values.

To use the relationship described above, we'll need to (unless if you are thinking about any alternatives):
1. grab the text contents of required sections in `html` at runtime
2. format those text contents to camel case (e.g., `accessToAndUseOfContent`),
3. associate them with the original `md` file (not the `html` page generated using the `md` file) and runtime generated `path` with `id` for  the `html` page (`id` won't be added to the `md` file)

Should we create a helper for these steps instead of manual editing (which might be error-prone)? And since the `md` file is the source of truth, should we validate the resulting `html` page?

About the static checking for `path`:
> This check can be performed statically. We should set this up so that we can update the config and run tests to ensure that a) one match exists, and b) that there is only one match.

<code>path: '/terms-of-use<strong>#access-to-and-use-of-content'</strong></code>

Currently the `id` (e.g., **`access-to-and-use-of-content`**) will be added to `html` after the page loads (i.e., after the `md` file has been converted to `html` at runtime). Since only after the entire page is ready (the content of `md` file in `html` format), we can use JS DOM method to traverse DOM elements, locate the required sections, and auto-insert the `id` name into them using the corresponding text content.

Could you please let me know the approach to statically test the value of the `path` using the `id` generated at runtime and the `md` file? Do you have any plugins or other tools in mind that we could utilize for FE's static testing (or would it be done outside of FE)?

Reply:
This was updated, please see the comment https://github.com/AlexsLemonade/scpca-portal/pull/555#discussion_r1508297208.